tests/common.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Contains several commonly used objects (models, dataloaders) that are shared across the test suite."""
+
+from typing import Sequence
+
+import pytest
+import torch
+from PIL import Image
+from torch.utils.data import Dataset
+from torchvision.datasets import VisionDataset
+
+from composer.core.precision import Precision
+from composer.models import ComposerClassifier
+
+# syntactic sugar below
+# decorators for common parameterizations and marks
+
+
+def device(*args, precision=False):
+    """Decorator for device and optionally precision.
+
+    Input choices are ('cpu', 'gpu'), or if precision=True,
+    also accept ('gpu-amp', 'gpu-fp32', and 'cpu-fp32').
+
+    Returns the parameter "device", or if precision=True,
+    also returns the parameter "precision".
+    """
+    # convert cpu-fp32 and gpu-fp32 to cpu, gpu
+    if not precision and any(['-' in arg for arg in args]):
+        raise ValueError('-fp32 and -amp tags must be removed if precision=False')
+    args = [arg.replace('-fp32', '') for arg in args]
+
+    # use lambdas to construct DeviceGPU only when gpu requested
```

Comment:
Can you elaborate? I don't think DeviceGPU needs to get registered for the decorator?